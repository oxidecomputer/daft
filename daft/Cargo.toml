[package]
name = "daft"
version = "0.1.0"
description = "Structural diffs of Rust data structures"
edition.workspace = true
repository.workspace = true
rust-version.workspace = true
license.workspace = true

[lints]
workspace = true

[dependencies]
daft-derive.workspace = true
newtype-uuid = { workspace = true, optional = true }
oxnet = { workspace = true, optional = true }
paste.workspace = true
uuid = { workspace = true, optional = true, features = ["v4"] }

[features]
newtype-uuid1 = ["dep:newtype-uuid"]
oxnet01 = ["dep:oxnet"]
uuid1 = ["dep:uuid"]
