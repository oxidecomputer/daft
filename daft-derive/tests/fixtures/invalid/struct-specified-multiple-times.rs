use daft::Diffable;

#[derive(Diffable)]
#[daft(leaf, leaf, leaf)]
struct MyStruct {
    a: i32,
    b: String,
}

#[derive(Diffable)]
#[daft(leaf)]
#[daft(leaf)]
#[daft(leaf)]
struct MyStruct2 {
    a: i32,
    b: String,
}

fn main() {
    // MyStruct and MyStruct2 should still exist, even though the Diffable impl
    // couldn't be generated.
    let _ = MyStruct { a: 0, b: "foo".to_string() };
    let _ = MyStruct2 { a: 0, b: "foo".to_string() };
}
