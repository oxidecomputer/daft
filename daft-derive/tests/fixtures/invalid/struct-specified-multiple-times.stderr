error: #[daft(leaf)] specified multiple times
 --> tests/fixtures/invalid/struct-specified-multiple-times.rs:4:14
  |
4 | #[daft(leaf, leaf, leaf)]
  |              ^^^^

error: #[daft(leaf)] specified multiple times
 --> tests/fixtures/invalid/struct-specified-multiple-times.rs:4:20
  |
4 | #[daft(leaf, leaf, leaf)]
  |                    ^^^^

error: #[daft(leaf)] specified multiple times
  --> tests/fixtures/invalid/struct-specified-multiple-times.rs:12:8
   |
12 | #[daft(leaf)]
   |        ^^^^

error: #[daft(leaf)] specified multiple times
  --> tests/fixtures/invalid/struct-specified-multiple-times.rs:13:8
   |
13 | #[daft(leaf)]
   |        ^^^^
